public/app/features/variables/state/batchStateReducer.ts

```diff
@@ -0,0 +1,41 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+export enum BatchStatus {
+  NotStarted = 'Not started',
+  Fetching = 'Fetching',
+  Completed = 'Completed',
+}
+
+export interface BatchState {
+  uid: string | undefined | null;
+  status: BatchStatus;
+}
+
+export const initialBatchState: BatchState = { uid: null, status: BatchStatus.NotStarted };
+
+const batchStateSlice = createSlice({
```

Comment:
I think this PR looks really good. The only think that comes to my mind is a really small nit when it comes to the naming. I would like to call this something like transaction. Feels more like a transaction then a batch. What do you think about that?

Apart from that. Well done!!

Reply:
Good suggestion, will change that!